lib/src/Base/Func/openturns/LinearEnumerateFunction.hxx

```diff
@@ -71,6 +71,9 @@ public:
   /** Upper bound accessor */
   void setUpperBound(const Indices & upperBound) override;
 
+  /** The marginal enumerate function */
+  LinearEnumerateFunction getMarginal(const Indices & indices) const;
```

Comment:
it should return an EnumerateFunction
and methods in child classes must overload it with the same signature (use virtual/override in parent/child)
else this wont be usable from c++

Reply:
I already tried, but it fails. Notice that, before implementing it this way, everything builds and runs:

![image](https://github.com/openturns/openturns/assets/31351465/472b273b-7048-413f-a7c2-1974210ecd57)

Then the commit 004f56c26 implements it, as you suggested. It builds fine. But the execution of the unit test `t_LinearEnumerateFunction_std` fails at runtime:
```
$ ./lib/test/t_LinearEnumerateFunction_std
terminate called after throwing an instance of 'OT::NotYetImplementedException'
  what():  In EnumerateFunctionImplementation::operator()
Abandon
```
Here is why. The `getMarginal()` method now returns an `EnumerateFunctionImplementation`. This class does not have any evaluation operator: only derived classes have.

I guess that this requires a dedicated code to make it work. I suspect that this is precisely the reason why [MarginalEvaluation.cxx](https://github.com/openturns/openturns/blob/master/lib/src/Base/Func/MarginalEvaluation.cxx) class is built for. The constructor of this class takes a function and a list of indices and builds the marginal function based on the output indices using pointer trickeries (see [here](https://github.com/openturns/openturns/blob/b5fd8338850d1255233f64a0e0c27d963d6f7daf/lib/src/Base/Func/MarginalEvaluation.cxx#L43)). Then the evaluation operator evaluates the function (see [here](https://github.com/openturns/openturns/blob/b5fd8338850d1255233f64a0e0c27d963d6f7daf/lib/src/Base/Func/MarginalEvaluation.cxx#L119)).

The method used in `MarginalEvaluation` cannot be applied to the `EnumerateFunction`s. This is because the method to compute each marginal linear enumeration rule is specific to each class. For example, the `HyperbolicAnisotropicEnumerationRule` takes into account for the list of marginal weights. There are not weights involved in the other enumeration rules (e.g. LinearEnumerationRule).

All in all, these are the reasons why I removed this commit. I temporarily let it in the history so that you can see and analyze the issue by yourself. Please let me know if you find another way to implement this.